ush/sfs_atmos_daily_monthly.sh

```diff
@@ -0,0 +1,186 @@
+#!/bin/bash
+
+######################################################################################
+
+# GENERATE MONTHLY/DAILY MEAN GRIB2 FILES ON INTERPOLATED 360-x181 LAT-LON  GRID 
+# FOR SFS MASTER DATA FILES
+
+#####################################################################################
+
+firstfile=$memdir/sfs*master.grb2f000
+
+# get validation date of first file
+vt_init=`wgrib2 $firstfile -d 1 -vt`
+mm_init=`echo $vt_init | cut -c12-13`
+
+#### Set index for finding month of validation date for loops
+months_in_year=("1" "2" "3" "4" "5" "6" "7" "8" "9" "10" "11" "12")
+start_idx=$(($(($mm_init))-1))
+
+#### check for leap year
+for i in {1..12}
+do
+  itime=$(wgrib2 -t $firstfile|head -1|cut -d= -f2)
+  yyyy=${itime:0:4}
+  mm=${itime:4:2}
+
+  ndays=$(date -d "$yyyy-$mm-01 +$i month -1 day" "+%d")
+
+  if (( "$ndays"==28 )); then
+    month_days_in_year=("31" "28" "31" "30" "31" "30" "31" "31" "30" "31" "30" "31")
+    break
+  fi
+  if (( "$ndays"==29 )); then
+    month_days_in_year=("31" "29" "31" "30" "31" "30" "31" "31" "30" "31" "30" "31")
+    break
+  fi 
+done
```

Comment:
Seems a bit excessive, looping over the same `firstfile` for 12 months
```suggestion
itime=$(wgrib2 -t $firstfile|head -1|cut -d= -f2)
yyyy=${itime:0:4}

if (( yyyy % 4 == 0 )); then # this is a leap year
  feb="29"
else
  feb="28"
fi
month_days_in_year=("31" "${feb}" "31" "30" "31" "30" "31" "31" "30" "31" "30" "31")
```

Reply:
@aerorahul Thanks for the suggestion. The problem is that the yyyy from the first file in this case is 2012 (the forecast starts in November), which is a leap year. However, the February month of this forecast data is from 2013, so the month has 28 days, not a leap year. Can you suggest how to account for that without having to loop through the files to find the year of the February files?

Reply:
@KarinaAsmar-NOAA 
Thanks for explaination.  I will retract my comment, as I don't yet fully understand this program to provide feedback.

Reply:
No problem, thank you for the suggestion. @JesseMeng-NOAA if you know of a quicker way to find a leap year, please let me know.